runtime/sam/expr/function/types.go

```diff
@@ -31,6 +31,12 @@ type NameOf struct {
 
 func (n *NameOf) Call(_ zed.Allocator, args []zed.Value) zed.Value {
 	typ := args[0].Type()
+	if typ.ID() == zed.IDType {
+		var err error
+		if typ, err = n.zctx.LookupByValue(args[0].Bytes()); err != nil {
+			panic(err)
+		}
+	}
 	if named, ok := typ.(*zed.TypeNamed); ok {
 		return zed.NewString(named.Name)
 	}
```

Comment:
This changes the output of
```
echo '<int64>(=a) <b=int64> <c=int64>(=d)' | zq 'yield nameof(this)' -
```
from
```
"a"
error("missing")
"d"
```
to
```
error("missing")
"b"
"c"
```
but I think we ultimately want
```
"a"
"b"
"d"
```

More explicitly, I propose that the `nameof()` algorithm be this:
1. If the argument is a named type, return its name.
2. Otherwise, if the argument is a type value that encodes a named type, return its name.
3. Otherwise, return `error("missing")`.

Reply:
Good point didn't consider the case of a named type with the value of type, dawg.